p2p/host/basic/basic_host.go

```diff
@@ -142,10 +148,21 @@ func NewHost(ctx context.Context, net network.Network, opts *HostOpts) (*BasicHo
 	if h.emitters.evtLocalProtocolsUpdated, err = h.eventbus.Emitter(&event.EvtLocalProtocolsUpdated{}); err != nil {
 		return nil, err
 	}
-	if h.emitters.evtLocalAddrsUpdated, err = h.eventbus.Emitter(&event.EvtLocalAddressesUpdated{}); err != nil {
+	if h.emitters.evtLocalAddrsUpdated, err = h.eventbus.Emitter(&event.EvtLocalAddressesUpdated{}, eventbus.Stateful); err != nil {
```

Comment:
I'm not sure if this should be stateful. Are we doing this to get around the fact that we don't have some form of `GetPeerRecord()` method as described in https://github.com/libp2p/go-libp2p/pull/747#issuecomment-596537472? If so, we should just add that method.

Reply:
@Stebalien We have a `GetPeerRecord(p peer.ID)` on the peerstore now.

However, calling that when you get a `EvtLocalAddressChanged` event could be racy as the address you see on the event could be different fro the signed record you get from the peerstore.

This avoids that race.